gradio/components/login_button.py

```diff
@@ -96,21 +96,11 @@ def _check_login_status(self, request: Request) -> LoginButton:
 
 
 # JS code to redirects to /login/huggingface if user is not logged in.
-# If the app is opened in an iframe, open the login page in a new tab.
-# Otherwise, redirects locally. Taken from https://stackoverflow.com/a/61596084.
-# If user is logged in, redirect to logout page (always in-place).
+# If user is logged in, redirect to /logout page. Always happens
+# on the same tab.
 _js_handle_redirect = """
 (buttonValue) => {
-    if (buttonValue === BUTTON_DEFAULT_VALUE) {
-        url = '/login/huggingface' + window.location.search;
-        if ( window !== window.parent ) {
-            window.open(url, '_blank');
-        } else {
-            window.location.assign(url);
-        }
-    } else {
-        url = '/logout' + window.location.search
-        window.location.assign(url);
-    }
+    uri = buttonValue === BUTTON_DEFAULT_VALUE ? '/login/huggingface' : '/logout';
+    window.location.assign(uri + window.location.search);
```

Comment:
This is where we will need to set the scrolling of the iframe as mention in @coyotte508's PR.

We should be able to add this as the first line of this function:
```js
window.parent?.postMessage({type: "SET_SCROLLING", enabled: true})
```